src/ATen/native/xpu/sycl/ReplicationPaddingKernels.cpp

```diff
@@ -0,0 +1,737 @@
+#pragma clang diagnostic push
+#pragma GCC diagnostic push
+// Avoid SYCL compiler return-type error
+#pragma clang diagnostic ignored "-Wreturn-type"
+#pragma GCC diagnostic ignored "-Wreturn-type"
+
+#include <ATen/ATen.h>
+#include <ATen/Context.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/IndexingUtils.h>
+#include <ATen/ceil_div.h>
+#include <ATen/native/xpu/sycl/Atomics.h>
+#include <comm/Runtime.h>
+#include <comm/SYCLContext.h>
+
+namespace at::native::xpu {
+
+inline int imin(int a, int b) {
+  return a > b ? b : a;
+}
+inline int imax(int a, int b) {
+  return a > b ? a : b;
+}
+
+template <typename scalar_t, typename F>
+struct ParallelReplicationPad1dKernelFunctor {
+  void operator()(sycl::nd_item<3> item) const {
+    auto output_id = item.get_global_id(2);
+    if (output_id < output_plane_size_) {
+      int64_t output_x = output_id % output_.size(2);
+      int64_t i_start_x = imax(0, -pad_left_);
+      int64_t o_start_x = imax(0, pad_left_);
+      int64_t input_x =
+          imin(imax(pad_left_, output_x), input_.size(2) + pad_left_ - 1) -
+          o_start_x + i_start_x;
+
+      f_(input_, output_, item.get_group(1), item.get_group(0), output_x, input_x);
+    }
+  }
+  ParallelReplicationPad1dKernelFunctor(
+      PackedTensorAccessor64<scalar_t, 3> input,
+      PackedTensorAccessor64<scalar_t, 3> output,
+      int64_t pad_left,
+      int64_t pad_right,
+      const F f,
+      int64_t output_plane_size)
+      : input_(input),
+        output_(output),
+        pad_left_(pad_left),
+        pad_right_(pad_right),
+        f_(f),
+        output_plane_size_(output_plane_size) {}
+
+ private:
+  PackedTensorAccessor64<scalar_t, 3> input_;
+  PackedTensorAccessor64<scalar_t, 3> output_;
+  int64_t pad_left_;
+  int64_t pad_right_;
+  const F f_;
+  int64_t output_plane_size_;
+};
+
+template <typename scalar_t, typename F>
+void parallel_replication_pad1d(
+    PackedTensorAccessor64<scalar_t, 3> input,
+    PackedTensorAccessor64<scalar_t, 3> output,
+    int64_t pad_left,
+    int64_t pad_right,
+    const F& f) {
+  auto queue = getCurrentSYCLQueue();
+  int64_t output_plane_size = output.size(2);
+
+  ParallelReplicationPad1dKernelFunctor<scalar_t, F> kfn(
+      input, output, pad_left, pad_right, f, output_plane_size);
+
+  int64_t work_group_size = syclMaxWorkGroupSize(kfn);
+  int64_t work_group_num = at::ceil_div(output_plane_size, work_group_size);
+  int64_t nplane = output.size(1);
+  int64_t nbatch = output.size(0);
+
+  sycl_kernel_submit(
+      sycl::range<3>(nbatch, nplane, work_group_size * work_group_num),
+      sycl::range<3>(1, 1, work_group_size),
+      queue,
+      kfn);
+}
+
+template <typename scalar_t>
+struct replication_pad1d_forward_kernel_functor {
```

Comment:
Please rename all kernel related structures,
1. `KernelFunctor` is used to name kernel frame.
2. The scalar functor is named by like `ReplicationPad1dForwardFunctor`. W/o kernel in the name.